OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs

```diff
@@ -89,59 +100,25 @@ public Order IssueOrder(Actor self, IOrderTargeter order, Target target, bool qu
 
 		public void ResolveOrder(Actor self, Order order)
 		{
-			if (order.OrderString != "DeployToUpgrade")
+			if (order.OrderString != "DeployToUpgrade" || deployState == DeployState.Deploying)
 				return;
 
 			if (!IsOnValidTerrain())
 				return;
 
-			if (isUpgraded)
+			if (deployState == DeployState.Deployed)
 			{
-				// Play undeploy animation and after that revoke the upgrades
-				self.QueueActivity(false, new CallFunc(() =>
-				{
-					if (!string.IsNullOrEmpty(info.UndeploySound))
-						Game.Sound.Play(info.UndeploySound, self.CenterPosition);
-
-					if (string.IsNullOrEmpty(info.DeployAnimation))
-					{
-						RevokeUpgrades();
-						return;
-					}
-
-					if (body.Value != null)
-						body.Value.PlayCustomAnimationBackwards(self, info.DeployAnimation, RevokeUpgrades);
-					else
-						RevokeUpgrades();
-				}));
+				self.CancelActivity();
```

Comment:
This will behave badly if the order is queued.  I think at a minimum this needs to be pulled above the if, and live in its own `if (!order.Queued)`.  @obrakmann may be able to find some other ways to break this based on our recent experience with the `Transforms` trait.